### PR TITLE
Single file analysis

### PR DESCRIPTION
**Pull request by ReshmaSobhaNair**

### CODE REVIEW

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 1]
```diff
@@ -185,6 +210,43 @@ protected Optional<? extends AbstractClassSource<JavaSootClass>> getClassSourceI
...
+   @Override
+   @Nonnull
+    public Optional<? extends AbstractClassSource<JavaSootClass>> getClassSource(@Nonnull ClassType type, @Nonnull View<?> view) {
+      return getClassSourceInternal((JavaClassType) type, path, new AsmJavaClassProvider(view));
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends AbstractClassSource<JavaSootClass>> getClassSources(@Nonnull View<?> view) {
+      return null;
```
Comment (swissiety) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java]: please return the single class in a `Collection.singletonList(...)` as well

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 2]
```diff
@@ -185,6 +210,43 @@ protected Optional<? extends AbstractClassSource<JavaSootClass>> getClassSourceI
...
+    public ClassFileBasedAnalysisInputLocation(Path classFilePath) {
+      super(classFilePath);
+      if (!classFilePath.toString().endsWith(".class")) {
+        throw new IllegalArgumentException("Provided path is not a .class file.");
+      }
+    }
+
+    public boolean doesClassExist() {
+      return Files.exists(path) && Files.isRegularFile(path);
+    }
```
Comment (swissiety) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java]: please check that in the constructor by default - else throw an IAE

File: sootup.core/src/main/java/sootup/core/inputlocation/FileType.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public enum FileType {
   WAR("war"),
   JIMPLE("jimple");
 
-  public static final @Nonnull EnumSet<FileType> ARCHIVE_TYPES = EnumSet.of(JAR, ZIP, APK, WAR);
+  public static final @Nonnull EnumSet<FileType> ARCHIVE_TYPES = EnumSet.of(JAR, ZIP, APK, WAR,CLASS);
```
Comment (swissiety) [inline sootup.core/src/main/java/sootup/core/inputlocation/FileType.java]: why is CLASS in the archive_types?

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 4]
```diff
@@ -185,6 +219,32 @@ protected Optional<? extends AbstractClassSource<JavaSootClass>> getClassSourceI
     return classProvider.createClassSource(this, pathToClass, signature);
   }
 
+  private static class ClassFileBasedAnalysisInputLocation extends PathBasedAnalysisInputLocation {
+    public ClassFileBasedAnalysisInputLocation(
+        @Nonnull Path classPath, @Nonnull SourceType srcType) {
+      super(classPath, srcType);
```
Comment (kadirayk) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java:203]: can you please check here if the file exists and throw an exception if not.

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 5]
```diff
@@ -494,7 +557,7 @@ private String dex2jar(Path path) {
       int start = apkPath.lastIndexOf(File.separator);
       int end = apkPath.lastIndexOf(".apk");
       String outputFile = outDir + apkPath.substring(start + 1, end) + ".jar";
-      Dex2jarCmd.main("-f", apkPath, "-o", outputFile);
+      // Dex2jarCmd.main("-f", apkPath, "-o", outputFile);
```
Comment (stschott) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java]:
```suggestion
      Dex2jarCmd.main("-f", apkPath, "-o", outputFile);
```

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 6]
```diff
@@ -4,7 +4,6 @@
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.google.common.cache.RemovalNotification;
-import com.googlecode.dex2jar.tools.Dex2jarCmd;
 import java.io.*;
```
Comment (stschott) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java]:
```suggestion
import com.googlecode.dex2jar.tools.Dex2jarCmd;
import java.io.*;
```

**Review by kadirayk — APPROVED** [review]

**Pull request merged**